src/routes/asset.php

```diff
@@ -199,8 +199,8 @@
         if (!isset($previews[$i]['thumbnail']) || $previews[$i]['thumbnail'] == '') {
             if ($previews[$i]['type'] == 'video') {
                 $matches = [];
-                if (preg_match('|youtube.com/watch\\?v=([^&]+)|', $previews[$i]['link'], $matches)) {
-                    $previews[$i]['thumbnail'] = 'http://img.youtube.com/vi/'.$matches[1].'/default.jpg';
+                if (preg_match('|youtube\.com/watch\\?v=([^&]+)|', $previews[$i]['link'], $matches)) {
```

Comment:
Pretty sure you need two slashes, as in `\\.`

Reply:
Works fine locally (PHP 7.3)
```php
function test($url) {
    if (preg_match('|youtube\.com/watch\\?v=([^&]+)|', $url, $matches)) {
        echo 'https://img.youtube.com/vi/'.$matches[1].'/default.jpg' . PHP_EOL;
    } else {
        echo 'no match' . PHP_EOL;
    }
}

test('https://www.youtube.com/watch?v=lakdfjladfk');
test('https://www.youtube§com/watch?v=lakdfjladfk');
```
```
$ php youtube.php 
https://img.youtube.com/vi/lakdfjladfk/default.jpg
no match
```
Should work in PHP 5 as well I think.

Reply:
Works in every PHP version: https://3v4l.org/5l7mX.
So should I remove the second `\` in `\\?` instead? :D

Reply:
https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.single

Seems like either is fine, in this programming language. Feel free to either remove one or add the other, just make it consistent :smiley:

Reply:
Done. The [preg_match examples](https://www.php.net/manual/en/function.preg-match.php#example-6035) use only one backslash as well, so I went with that. Now we have it the official PHP way™. :grimacing: